File: apps/ruby-sdk/lib/firecrawl/models/batch_scrape_response.rb
# frozen_string_literal: true

module Firecrawl
  module Models
    # Response from starting an async batch scrape job.
    class BatchScrapeResponse
      attr_reader :id, :url, :invalid_urls

      def initialize(data)
        @id = data["id"]
        @url = data["url"]
        @invalid_urls = data["invalidURLs"]
      end

      def to_s
        "BatchScrapeResponse{id=#{id}, url=#{url}}"
      end
    end
  end
end

File: apps/dot-net-sdk/Firecrawl/Models/BatchScrapeResponse.cs
using System.Text.Json.Serialization;

namespace Firecrawl.Models;

/// <summary>
/// Response from starting an async batch scrape job.
/// </summary>
public class BatchScrapeResponse
{
    [JsonPropertyName("success")]
    public bool Success { get; set; }

    [JsonPropertyName("id")]
    public string? Id { get; set; }

    [JsonPropertyName("invalidURLs")]
    public List<string>? InvalidURLs { get; set; }
}
